admin/preprints/views.py

```diff
@@ -380,19 +381,29 @@ def delete(self, request, *args, **kwargs):
             raise PermissionDenied('You do not have permission to update a preprint flagged as spam.')
         preprint_ids = [
             pid for pid in request.POST.keys()
-            if pid != 'csrfmiddlewaretoken'
+            if pid not in ('csrfmiddlewaretoken', 'spam_confirm', 'ham_confirm')
         ]
         for pid in preprint_ids:
             preprint = Preprint.load(pid)
             osf_admin_change_status_identifier(preprint)
-            preprint.confirm_spam(save=True)
-            update_admin_log(
-                user_id=self.request.user.id,
-                object_id=pid,
-                object_repr='Preprint',
-                message='Confirmed SPAM: {}'.format(pid),
-                action_flag=CONFIRM_SPAM
-            )
+            if ('spam_confirm' in list(request.POST.keys())):
+                preprint.confirm_spam(save=True)
+                update_admin_log(
+                    user_id=self.request.user.id,
+                    object_id=pid,
```

Comment:
nit: No need to check the request keys in the loop. This should probably be checked at the same time the preprint ids are checked.

Also, I'm assuming that this function is only called by a caller that *knows* it's confirming spam/ham, otherwise we should handle the case where neither supported key is present and/or where both are.